File: Year 1/Semester 1/Java/Lessons/nested_states.java
import java.util.Scanner;

public class nested_states{

    public static void main(String[] args) {

        System.out.println("Input your age: ");

        Scanner entry = new Scanner(System.in);
        String s = entry.nextLine();

        int age = Integer.parseInt(s);

        if (age >= 21) {
            System.out.println("Input your favourite food: ");
            String food = entry.nextLine();

            if (food.equals("pizza")) {
                System.out.println("nice");
            }
            else {
                System.err.println("Not pizza i guess");
            }
        }
        else if (age > 13 && age < 21) {
            System.out.println("You are a teen");
            
        }
        else {
            System.out.println("You are not a teen or an adult");
        }


    }
}

File: Year 1/Semester 1/C/Lessons/nested_states.c
#include <stdio.h>

int main() {
    int age;
    char food[50]; // Array to store the favorite food input

    printf("Input your age: ");
    scanf("%d", &age); // Read age from user input

    if (age >= 21) {
        printf("Input your favourite food: ");
        scanf("%s", food); // Read favorite food from user input

        // Check if the favorite food is pizza
        if (strcmp(food, "pizza") == 0) {
            printf("nice\n");
        } else {
            printf("Not pizza I guess\n");
        }
    } else if (age > 13 && age < 21) {
        printf("You are a teen\n");
    } else {
        printf("You are not a teen or an adult\n");
    }

    return 0; // Indicate successful completion
}
